### PR TITLE
Add python package dependencies to setup.py

### PR DESCRIPTION
**Pull request by josepht**
* Make bin/snapcraft-parser runnable from a virtualenv.

LP:#1603997
LP:#1568901

### CODE REVIEW

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by fallen** [comment]
I can confirm the following requirements are missing in setup.py to allow installation:
missing requirements for snapcraft:

jsonschema
file-magic
yaml
python-apt
py3-progressbar
requests-toolbelt
docopt
petname

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

File: setup.py [hunk 1]
```diff
@@ -50,10 +50,26 @@
...
+        'file-magic',
+        'jsonschema',
+        'oauthlib',
+        'pkg-resources',
+        'py3-progressbar',
+        'pyxdg',
+        'PyYAML',
         'pyxdg',
-        'requests',
+        'Requests',
```
Comment (squidsoup) [inline setup.py]: Reasonably certain this should be 'requests':

`error: The 'Requests' distribution was not found and is required by snapcraft`

Comment (josepht) [inline setup.py]: Which release of Ubuntu are you running into this on?

On Xenial I get:
error: The 'requests' distribution was not found and is required by snapcraft

I get that the first time I run './setup.py install' but the next call succeeds.

File: setup.py [hunk 2]
```diff
@@ -50,10 +50,26 @@
         ('share/snapcraft/libraries',
             ['libraries/' + x for x in os.listdir('libraries')]),
     ],
+    dependency_links=[
+        'git+git://anonscm.debian.org/apt/python-apt.git#egg=python_apt-0.0.0',
```
Comment (sergiusens) [inline setup.py]: isn't there an https version of this?

Comment (josepht) [inline setup.py]: I found one and pushed an update.

**Comment by sergiusens** [comment]
ok to test

**Comment by ArmstrongJ** [comment]
Trying to install snapcraft and guessing at these requirements is not pleasant.  This pull request really needs to be accepted...

**Comment by sergiusens** [comment]
El 05/07/16 a las 17:42, Jeff Armstrong escribió:
> Trying to install snapcraft and guessing at these requirements is not
> pleasant. This pull request really needs to be accepted...
It causes a strange side effect when debuilding, reason for which it hasn't been accepted yet.

Maybe moving it to a requirements.txt is a good option

**Comment by sergiusens** [comment]
sorry to rain on your parade, but tabulate is missing.

@elopio can we use this for docker unit testing?

**Comment by elopio** [comment]
@sergiusens let's see: #656

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,20 @@
+configparser==3.5.0
+docopt==0.6.2
+file-magic==0.3.0
+jsonschema==2.5.1
+oauthlib==1.0.3
+pkg-resources==0.0.0
```
Comment (elopio) [inline requirements.txt]: why is this one 0.0.0?

Comment (josepht) [inline requirements.txt]: That package is a dependency of one of the packages we are installing from a tarball and is installed automatically.  I've removed it from requirements.txt.

**Comment by elopio** [comment]
@sergiusens @josepht #656 is ready, with an ugly hack to get the coverage working from deb.
:+1: for me here.

**Comment by sergiusens** [comment]
What's the bug number for this?

**Comment by josepht** [comment]
I've added the bug numbers to the PR description.  Do you need me to add them to the initial commit message as well?

**Pull request merged**